### PR TITLE
Add sensible setup for vscode

### PR DESCRIPTION
**Pull request by fzakaria**
* Add CmakePresets.json file with the default Debug build. This actually works for Cmake itself without vscode.

  `cmake --preset debug`

* add  `ide.md` to describe  how to setup vscode to get intellisense and to use the CMakePresets.json file

End result of this should be that CMake works nicely in vscode **with** clangd intellsense information

image

### CODE REVIEW

File: build_tools/github_actions/lint_check_license.sh [hunk 1]
```diff
@@ -59,6 +59,7 @@ SKIPPED_SUFFIXES=(
   LICENSE
   MODULE.bazel.lock
   llvm_version.txt
+  .json
```
Comment (mlevesquedion) [inline build_tools/github_actions/lint_check_license.sh:62]: Nit: insert above to maintain sorted order?

**Comment by fzakaria** [comment]
@mlevesquedion did you try it ?
I'm keen to see if the setup works nicely for others.

I think at a minimum we can contribute the preset file and then remove the large CMake command in our README.

**Comment by mlevesquedion** [comment]
> @mlevesquedion did you try it ? I'm keen to see if the setup works nicely for others.
> 
> I think at a minimum we can contribute the preset file and then remove the large CMake command in our README.

I use Vim and I'm happy with it. I tried VSCode at one point but it wasn't as nice.

I like the CMake preset idea though!

**Comment by fzakaria** [comment]
Oh yeah -- I forgot your vim-guru.
I'll wait until someone else who wants to test VSCode to give it a go so I have another signal.

**Comment by abhigunj** [comment]
I will validate on vscode

**Comment by fzakaria** [comment]
@abhigunj tagged you on it since we went together.
moved the instructions to a IDE.md

**Comment by abhigunj** [comment]
> @abhigunj tagged you on it since we went together. moved the instructions to a IDE.md

Thanks, instructions are clear and cover issues I faced. 
we also hit the missing ccache pkg but that is unrelated to this PR/changes.

**Review by abhigunj — APPROVED** [review]

**Pull request merged**